sycl/test-e2e/HostInteropTask/interop-task-cuda.cpp

```diff
@@ -0,0 +1,134 @@
+// RUN: %{build} -o %t.out -lcuda
+// RUN: %{run} %t.out
+// REQUIRES: cuda
+
+#include <iostream>
+#include <sycl/backend.hpp>
+#include <sycl/detail/core.hpp>
+#include <sycl/detail/host_task_impl.hpp>
+
+#include <cuda.h>
+
+using namespace sycl;
+using namespace sycl::access;
+
+static constexpr size_t BUFFER_SIZE = 1024;
+
+template <typename T> class Modifier;
+
+template <typename T> class Init;
+
+template <typename BufferT, typename ValueT>
+void checkBufferValues(BufferT Buffer, ValueT Value) {
+  auto Acc = Buffer.get_host_access();
+  for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+    if (Acc[Idx] != Value) {
+      std::cerr << "buffer[" << Idx << "] = " << Acc[Idx]
+                << ", expected val = " << Value << '\n';
+      exit(1);
+    }
+  }
+}
+
+template <typename DataT>
+void copy(buffer<DataT, 1> &Src, buffer<DataT, 1> &Dst, queue &Q) {
+  Q.submit([&](handler &CGH) {
+    auto SrcA = Src.template get_access<mode::read>(CGH);
+    auto DstA = Dst.template get_access<mode::write>(CGH);
+
+    auto Func = [=](interop_handle IH) {
+      auto HipStream = IH.get_native_queue<backend::ext_oneapi_cuda>();
+      auto SrcMem = IH.get_native_mem<backend::ext_oneapi_cuda>(SrcA);
+      auto DstMem = IH.get_native_mem<backend::ext_oneapi_cuda>(DstA);
+
+      if (cuMemcpyAsync(DstMem, SrcMem, sizeof(DataT) * SrcA.get_count(),
+                        HipStream) != CUDA_SUCCESS) {
+        throw;
+      }
+
+      if (cuStreamSynchronize(HipStream) != CUDA_SUCCESS) {
+        throw;
+      }
+
+      if (Q.get_backend() != IH.get_backend())
+        throw;
+    };
+    CGH.host_task(Func);
+  });
+}
+
+template <typename DataT> void modify(buffer<DataT, 1> &B, queue &Q) {
+  Q.submit([&](handler &CGH) {
+    auto Acc = B.template get_access<mode::read_write>(CGH);
+
+    auto Kernel = [=](item<1> Id) { Acc[Id] += 1; };
+
+    CGH.parallel_for<Modifier<DataT>>(Acc.get_count(), Kernel);
```

Comment:
// Deprecated
  size_t get_count() const;\
Returns the same value as size().

lets avoid deprecated methods in new tests.

Reply:
Thanks have changed.